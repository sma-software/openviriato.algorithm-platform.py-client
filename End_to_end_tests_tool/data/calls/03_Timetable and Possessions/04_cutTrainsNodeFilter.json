{
    "restResource": "/trains",
    "httpMethod": "GET",
    "urlParameters": null,
    "queryParameters": [
        {
            "key": "FromTime",
            "value": "2003-05-01T00%3A10%3A00",
            "optional": false
        },
        {
            "key": "ToTime",
            "value": "2003-05-01T00%3A11%3A30",
            "optional": false
        },
        {
            "key": "NodeFilter",
            "value": "10,99",
            "optional": true
        },
        {
            "key": "CutTrain",
            "value": "true",
            "optional": true
        }
    ],
    "requestBody": null,
    "CSharpMethodName": "GetTrainsCutToTimeRangeDrivingAnyNode(timeWindow, nodeIDs)",
    "CSharpMethodParameterTypes": [ "SMA.AlgorithmPlatform.AlgorithmInterface.AIDM.ITimeWindow", "IImmutableList{System.Int64}" ],
    "subcategory": "Get Trains"
}
