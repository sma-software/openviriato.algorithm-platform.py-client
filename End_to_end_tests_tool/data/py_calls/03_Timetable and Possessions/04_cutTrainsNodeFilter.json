{
  "PythonParameterFromJson" :
  [
    {
      "ParameterName": "time_window",
      "aidm_factory": "convert_json_to_time_window",
      "ObjectJson": {
        "from_time": "2003-05-01T00:10:00",
        "to_time": "2003-05-01T00:11:30"
      }
    },
    {
      "ParameterName" : "node_filter",
      "aidm_factory": "extract_first_dict_value",
      "ObjectJson":
      {
        "node_filter" : [ 10, 99 ]
      }
    }
  ]
}