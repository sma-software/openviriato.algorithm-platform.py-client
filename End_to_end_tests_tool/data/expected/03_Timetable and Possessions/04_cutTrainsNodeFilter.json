[{"ID":2060,"TrainPathNodes":[{"ID":1696,"SectionTrackID":null,"NodeID":10,"NodeTrackID":null,"FormationID":null,"ArrivalTime":"2003-05-01T00:10:00","DepartureTime":"2003-05-01T00:10:00","MinimumRunTime":null,"MinimumStopTime":"P0D","StopStatus":"CommercialStop","SequenceNumber":1}],"DebugString":"RVZH_1_1_J03 tt_(G)"}]